data/ai/lua/generic_recruit_engine.lua

```diff
@@ -650,11 +661,29 @@ return {
             local best_scores = {offense = 0, defense = 0, move = 0}
             local best_hex = recruit_data.recruit.best_hex
             local target_hex = recruit_data.recruit.target_hex
-            local distance_to_enemy, enemy_location
-            if target_hex[1] then
-                distance_to_enemy, enemy_location = AH.get_closest_enemy(target_hex, wesnoth.current.side, { viewing_side = 0 })
-            else
-                distance_to_enemy, enemy_location = AH.get_closest_enemy(best_hex, wesnoth.current.side, { viewing_side = 0 })
+
+            local reference_hex = target_hex[1] and target_hex or best_hex
+            local distance_to_enemy, enemy_location = AH.get_closest_enemy(reference_hex, wesnoth.current.side, { viewing_side = 0 })
+            -- If no enemy is on the map, then we first use closest enemy start hex,
+            -- and if the does not exist either, a location mirrored w.r.t the center of the map
+            if not enemy_location then
```

Comment:
minor spelling nit: "and if that does not exist either"

Reply:
Yeah, seen that myself already too.  I'll fix it.